signing/secp256k1/impl/src/main/java/tech/pegasys/signers/secp256k1/multikey/MultiKeySignerProvider.java

```diff
@@ -33,75 +36,95 @@
+      final Path rootDir, final FileSelector<PublicKey> configFileSelector) {
     final SigningMetadataTomlConfigLoader signingMetadataTomlConfigLoader =
         new SigningMetadataTomlConfigLoader(rootDir);
 
-    final AzureKeyVaultTransactionSignerFactory azureFactory =
-        new AzureKeyVaultTransactionSignerFactory(new AzureKeyVaultAuthenticator());
+    final AzureKeyVaultSignerFactory azureFactory =
+        new AzureKeyVaultSignerFactory(new AzureKeyVaultAuthenticator());
 
     final HashicorpSignerFactory hashicorpSignerFactory = new HashicorpSignerFactory(Vertx.vertx());
 
-    return new MultiKeyTransactionSignerProvider(
-        signingMetadataTomlConfigLoader, azureFactory, hashicorpSignerFactory);
+    return new MultiKeySignerProvider(
+        signingMetadataTomlConfigLoader, azureFactory, hashicorpSignerFactory, configFileSelector);
   }
 
-  public MultiKeyTransactionSignerProvider(
+  public MultiKeySignerProvider(
       final SigningMetadataTomlConfigLoader signingMetadataTomlConfigLoader,
-      final AzureKeyVaultTransactionSignerFactory azureFactory,
-      final HashicorpSignerFactory hashicorpSignerFactory) {
+      final AzureKeyVaultSignerFactory azureFactory,
+      final HashicorpSignerFactory hashicorpSignerFactory,
+      final FileSelector<PublicKey> configFileSelector) {
     this.signingMetadataTomlConfigLoader = signingMetadataTomlConfigLoader;
     this.azureFactory = azureFactory;
     this.hashicorpSignerFactory = hashicorpSignerFactory;
+    this.configFileSelector = configFileSelector;
   }
 
   @Override
-  public Optional<TransactionSigner> getSigner(final String address) {
+  public Optional<Signer> getSigner(final PublicKey publicKey) {
     return signingMetadataTomlConfigLoader
-        .loadMetadataForAddress(address)
+        .loadMetadata(configFileSelector.getSpecificConfigFileFilter(publicKey))
         .map(metadataFile -> metadataFile.createSigner(this));
   }
 
   @Override
-  public Set<String> availableAddresses() {
-    return signingMetadataTomlConfigLoader.loadAvailableSigningMetadataTomlConfigs().stream()
-        .map(metadataFile -> metadataFile.createSigner(this))
+  public Set<PublicKey> availablePublicKeys() {
+    return signingMetadataTomlConfigLoader
+        .loadAvailableSigningMetadataTomlConfigs(configFileSelector.getAllConfigFilesFilter())
+        .stream()
+        .map(
+            metadataFile -> {
+              final Signer signer = metadataFile.createSigner(this);
+              try {
+                if ((signer != null)
+                    && configFileSelector
+                        .getSpecificConfigFileFilter(signer.getPublicKey())
+                        .accept(Path.of(metadataFile.getFilename()))) {
+                  return signer;
+                }
+                return null;
+              } catch (final IOException e) {
+                LOG.warn("IO Exception raised while loading {}", metadataFile.getFilename());
```

Comment:
may I suggest changing the warning to:

`LOG.warn("IO Exception raised while loading {}:{}", metadataFile.getFilename(), e.getMessage(),e);

Reply:
you may :) Lovely.